cmd/setup/03.go

```diff
@@ -83,9 +84,17 @@ func (mig *FirstInstance) Execute(ctx context.Context) error {
 	mig.instanceSetup.CustomDomain = mig.externalDomain
 	mig.instanceSetup.DefaultLanguage = mig.DefaultLanguage
 	mig.instanceSetup.Org = mig.Org
+	// check if username is email style or else append @<orgname>.<custom-domain>
+	//this way we have the same value as before changing `UserLoginMustBeDomain` to false
+	if !mig.instanceSetup.DomainPolicy.UserLoginMustBeDomain && !strings.Contains(mig.instanceSetup.Org.Human.Username, "@") {
+		mig.instanceSetup.Org.Human.Username = mig.instanceSetup.Org.Human.Username + "@" + domain.NewIAMDomainName(mig.instanceSetup.Org.Name, mig.instanceSetup.CustomDomain)
+	}
```

Comment:
Do we append this directly to the username? Do we get rid of the loginname? Before we just concatenated right?

Reply:
if UserLoginMustBeDomain is false a username is automatically the loginname
in case the username in the setup step has no email style (which is the case for the defaults.yaml), we need to generate an email style username until https://github.com/zitadel/zitadel/pull/4134 is solved.
Solving it this way also makes it backwords compatible as usernames for the first user were generated the same way in the commands